File: codewars/6kyu/Find The Parity Outlier.js
function findOutlier(integers) {
  const isEvenCount = integers.slice(0, 3).filter(num => num % 2 === 0).length;

  let majorityEven = isEvenCount >= 2;

  for (let i = 0; i < integers.length; i++) {
    if (majorityEven && integers[i] % 2 !== 0) {
      return integers[i];
    } else if (!majorityEven && integers[i] % 2 === 0) {
      return integers[i];
    }
  }
}

console.log(findOutlier([2, 4, 0, 100, 4, 11, 2602, 36])); // 11

File: codewars/6kyu/Find The Parity Outlier.py
def find_outlier(integers):
    odds = []
    evens = []
    for i in integers:
        if i % 2 == 0:
            evens.append(i)
        else:
            odds.append(i)

    if len(odds) == 1:
        return odds[0]
    elif len(evens) == 1:
        return evens[0]
    return None


if __name__ == "__main__":
    find_outlier([2, 4, 0, 100, 4, 11, 2602, 36])  # 11
